### PR TITLE
Test cases for avi_model_nodes and istio_model_rel

### PR DESCRIPTION
**Pull request by Rishi-Sastry**
I've written a few test cases for the Istio_model_rel.go file and the avi_model_nodes.go file.
The drain_test.go file is for the Istio_model_rel.go file. 
Changes Made to the drain_test.go file:
- Added additional lines of code to the TestVSServiceCreate() method to add another gateway object.
- Added additional lines of code to the TestVSServiceDelete() method to accommodate for the added gateway object.
- Added a method to test the VSToGateway() method. This method tests the VSToGateway() method by obtaining an array of the gateways and compares it to an array of gateways obtained from the GetGatewaysForVS() method.
- Added a method to test the GetGatewayNamespace() method. This method checks to see if the namespaces for the associated gateways belongs to the default namespace or not.
- Added a method to test the SVCToGateway() method. This method tests the SVCToGateway() method by obtaining an array of gateways and compares it to an array of gateways obtained through the GetGatewaysforVS() method after iterating through the VS's for each of the services and then comparing the two arrays.
- Added a method to test the EPToGateway() method. Since the Ep's are like services, the same idea that was applied to test the SVCToGateway() method was used.
- Added a method to test the GatewayChanges() method. Modifications were made to a particular gateway object and then tests were run on it to make sure the method worked.
- Added a method to test the DrToGateway() method.Still have to run a few tests on it and hence the function has been placed in comments.

Changes Made to the avi_model_nodes.go file:
-Made 2 test methods. One to test the checksum method for AviVsNode, AviPoolGroupNode and the AviPoolNode. The other one is to test some of the boundary cases for the match criteria functions.

### CODE REVIEW

**Comment by sudswasavi** [comment]
@Rishi-Sastry Hey, there are travis failures reported on the PR. Could you pls check it?
`# github.com/avinetworks/servicemesh/pkg/istio/nodes [github.com/avinetworks/servicemesh/pkg/istio/nodes.test]
pkg/istio/nodes/avi_model_nodes_test.go:41:156: too few values in AviPoolNode literal
FAIL	github.com/avinetworks/servicemesh/pkg/istio/nodes [build failed]
Makefile:30: recipe for target 'test' failed
make: *** [test] Error 2
The command "make test" exited with 2.
`

File: pkg/istio/nodes/drain_test.go [hunk 1]
```diff
@@ -67,6 +72,212 @@ func TestVSServiceCreate(t *testing.T) {
...
+	check, gws2 := v.GetGatewaysForVS("vs_5")
+	if !check {
+		g.Expect(gateways).To(gomega.Equal(gws2))
+	}
+
+}
+
+func TestGetGatewayNamespace(t *testing.T) {
+	// Testing the GetGatewayNamespace() method for individual gateways
+	gatewaynsTest := GetGatewayNamespace("default", "gw_1")
```
Comment (sudswasavi) [inline pkg/istio/nodes/drain_test.go]: The idea is to have a call like this:

GetGatewayNamespace("default", "istio-system/gw_1") --> this should return you istio-system
GetGatewayNamespace("default", "gw_1") --> This should return you default

File: pkg/istio/nodes/drain_test.go [hunk 2]
```diff
@@ -67,6 +72,212 @@ func TestVSServiceCreate(t *testing.T) {
...
+
+	for _, service := range svcs {
+		gateways := SvcToGateway(service, "default")
+		flag, vs := istio_objs.SharedSvcLister().Service("default").GetSvcToVS(service)
+		if flag {
+			for _, vsName := range vs {
+				check, gateway := v.GetGatewaysForVS(vsName)
+				if check {
+					for _, gw := range gateway {
+						g.Expect(gateways).To(gomega.ContainElement(gw))
```
Comment (sudswasavi) [inline pkg/istio/nodes/drain_test.go]: This is a good test :)

File: pkg/istio/nodes/drain_test.go [hunk 3]
```diff
@@ -67,6 +72,212 @@ func TestVSServiceCreate(t *testing.T) {
...
+// 	// Obtaining a Map of the DR Names and versions to iterate over each one and get the associated Gateways
+// 	Drnames := d.GetAllDrNameVers()
+// 	for key := range Drnames {
+// 		// Obtaining a gateway object for each of the DR Names
+// 		gateways := DrToGateway(key,"default")
+// 		// Obtaining a service object for each of the DR Names. The gateways associated with the services must be the same for the gateways associated with the DR
+// 		flag,svcs := d.GetSVCMapping(key)
+// 		if flag {
+// 			for service := range svcs {
+// 				// Obtaining all the gateways associated with the service
```
Comment (sudswasavi) [inline pkg/istio/nodes/drain_test.go]: You can remove this, if it's not needed.

**Comment by adityats** [comment]
@Rishi-Sastry @sudswasavi Is this ready for merge?

**Comment by sudswasavi** [comment]
@adityats Nope. Waiting for Rishi to push a commit with the code review comments incorporated.

**Comment by sudswasavi** [comment]
@adityats Travis is broken as you can probably note above.
